src/index.js

```diff
@@ -0,0 +1,141 @@
+
+// funtion for marking done lists😎😊😉😎............
+
+listHolder.addEventListener('change', (event) => {
+  if (event.target.type === 'checkbox' && event.target.checked) {
+    event.target.nextElementSibling.style.textDecoration = 'line-through';
+    event.target.completed = true;
+  } else if (event.target.type === 'checkbox' && !event.target.checked) {
+    event.target.nextElementSibling.style.textDecoration = 'none';
+  }
+});
+
+if (localStorage.getItem('listArray')) {
+  listArray = JSON.parse(localStorage.getItem('listArray'));
+  UI.displayData();
+}
+
+listHolder.addEventListener('click', (event) => {
+  if (event.target.classList.contains('fa-ellipsis-v')) {
+    event.target.classList.remove('fa-ellipsis-v');
+    event.target.classList.add('fa-trash');
+    event.target.parentElement.parentElement.style.backgroundColor = '#f39c12';
+    const targeted = event.target.parentNode.previousElementSibling.lastElementChild;
+    targeted.contentEditable = 'true';
+  } else if (event.target.classList.contains('fa-trash')) {
+    const index = event.target.parentElement.parentElement.getAttribute('data-index');
+    listArray.splice(index, 1);
+    localStorage.setItem('listArray', JSON.stringify(listArray));
+    event.target.parentElement.parentElement.remove();
+  }
+});
+listHolder.addEventListener('input', (e) => {
+  if (e.target.closest('#discription')) {
+    const inputText = e.target.closest('#discription');
+    const taskElm = inputText.parentNode.parentNode;
+    const index = Array.prototype.indexOf.call(listHolder.children, taskElm);
+    listArray[index].discription = e.target.textContent;
+
+    localStorage.setItem('listArray', JSON.stringify(listArray));
+  }
+});
+
+window.addEventListener('load', () => {
+  if (localStorage.getItem('listArray')) {
+    listArray = JSON.parse(localStorage.getItem('listArray'));
+    UI.displayData();
+  }
+});
+/*
+lists.forEach((list) =>{
+   list.addEventListener('click', () => {
+    console.log('clicked');
+   })
+})
+
+  Editing the list *********inputText.addEventListener(
+      'input',
+      (e) => {
+        const parent = document.querySelector('.list-container');
+        const child = e.target.closest('.task-container');
+        saveData(editText(loadData(), parent, child, inputText.textContent));
+      },
+      false,
+    );
+*/
+
+if (localStorage.getItem('listArray')) {
+  listArray = JSON.parse(localStorage.getItem('listArray'));
+  UI.displayData();
+}
+
+// Function to remove completed items from the listArray and the DOM
+function removeCompleted() {
+  listArray = listArray.filter((item) => !item.completed);
+  localStorage.setItem('listArray', JSON.stringify(listArray));
+  UI.displayData();
+  // Select all elements with the completed class and remove them from the DOM
+  const completedElements = document.querySelectorAll('.completed');
+  completedElements.forEach((element) => element.remove());
+}
```

Comment:
- [ ] There is an error when removing an Item from the array. As you can see in the screenshot below I have `5` tasks
![image](https://user-images.githubusercontent.com/81932478/214896679-9aa8968d-c698-4adf-8925-df14242caa16.png)
When I delete task number `4` from the list, in the browser it disappears but in local storage it persists, and instead task number `1` is deleted, kindly fix this issue
![image](https://user-images.githubusercontent.com/81932478/214897386-afda6707-5f4c-48e6-a3c7-e2d479462467.png)